### PR TITLE
Frontend and onboarding tweaks

### PR DESCRIPTION
**Pull request by blt950**
Tasks are quite fragmented, so normal merge _without_ squash is prefered here. See commits for "changelog"
This makes up to v4.0.2 release

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,12 +30,13 @@ To setup your Docker instance simply follow these steps:
 3. Configure the environment variables as described in the [CONFIGURE.md](CONFIGURE.md)
 4. Run the container
 5. Setup a crontab outside the container to run `* * * * * docker exec --user www-data -i control-center php artisan schedule:run >/dev/null` every minute. This patches into the container and runs the required cronjobs.
-6. Bind the 8080 (HTTP) and/or 8443 (HTTPS) port to your reverse proxy or similar.
+6. To ensure that users will not need to login after each time you re-deploy/upgrade the container, enter the container and run `php artisan key:get`. Copy the key and set it as the `APP_KEY` environment variable in your docker configuration.
```
Comment (thor) [inline README.md:33]: quick suggestion: WDYT about having a line with `docker exec` that does the `php artisan key:get`? I can chuck in a suggestion later.

**Review by Adrian2k — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -27,17 +30,20 @@ If you don't want to use Docker, you need:
 To setup your Docker instance simply follow these steps:
 1. Pull the `ghcr.io/vatsim-scandinavia/control-center:v4` Docker image
 2. Setup your MySQL database (not included in Docker image)
-3. Configure the environment variables as described in the [CONFIGURE.md](CONFIGURE.md)
-4. Run the container
+3. Configure the environment variables as described in the [CONFIGURE.md](CONFIGURE.md#environment)
+4. Run the container and access it with `docker exec -it control-center bash`
+    - Run `php artisan migrate` to setup the database
+    - To ensure that users will not need to login after each time you re-deploy/upgrade the container, run `php artisan key:get`. Copy the key and set it as the `APP_KEY` environment variable in your docker configuration.
```
Comment (thor) [inline README.md:36]:
````suggestion
4. Start the container in the background.
5. Setup the database.
   ```sh
   docker exec -t control-center php artisan migrate
   ```
6. To ensure that users will not need to log in after each time you re-deploy or upgrade the container, you need to create and store an application key in your environment.
   ```sh
   docker exec -t control-center php artisan key:get
   ```
   Copy the key and set it as the `APP_KEY` environment variable in your Docker configuration.
````

WDYT?

File: UPGRADE.md [hunk 3]
```diff
@@ -16,13 +18,14 @@ Please note that users need to login again after this update, as it'll create a
 
 We now offer a Docker container that can be used to run Control Center, this is the recommended hosting now that ensures correct environement. If you still prefer to run it without Docker, this is still possible by cloning the repo and building the project manually.
 
-*Note: The included docker-compose.yaml is only for development purposes and should not be used in production.*
+In the instructions where we use `docker exec` we assume your container is named `control-center`. If you have named differently, please replace this.
```
Comment (thor) [inline UPGRADE.md:21]: hmm, wasn't this written in another file too? 👀

Comment (thor) [reply UPGRADE.md:21]: I think we could keep the Docker setup instructions outside of the upgrade, as it's not actually part of the upgrade: you don't _need_ to switch to it, and you can do it whenever you feel like it.

File: app/Console/Commands/KeyGet.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     * @var string
+     */
+    protected $signature = 'key:get';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Get the currenlt application key';
```
Comment (thor) [inline app/Console/Commands/KeyGet.php:21]:
```suggestion
    protected $description = 'Get the current application key';
```

File: app/Console/Commands/UserMakeAdmin.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+     * @var string
+     */
+    protected $signature = 'user:makeadmin';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Makes the specificed user an admin';
```
Comment (thor) [inline app/Console/Commands/UserMakeAdmin.php:24]:
```suggestion
    protected $description = 'Set the specified user as admin';
```
or
```suggestion
    protected $description = 'Make the specified user into an admin';
```

File: app/Console/Commands/UserMakeAdmin.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    public function handle()
+    {
+
+        // Input user
+        $cid = $this->ask("What is the user's CID?");
+        if ($user = User::find($cid)) {
+            $this->info('User found: ' . $user->name);
+
+            $areas = Area::all();
+            $area = $this->choice('Which area? Practically admin has all access, but an area must be selected regardless.', $areas->pluck('name')->toArray());
```
Comment (thor) [inline app/Console/Commands/UserMakeAdmin.php:40]:
```suggestion
            $area = $this->choice('Which area? The admin has access across areas, but an area must be selected regardless.', $areas->pluck('name')->toArray());
```

File: config/app.php [hunk 7]
```diff
@@ -25,7 +25,7 @@
     |
     */
 
-    'version' => '4.0.1',
+    'version' => '4.0.2',
```
Comment (thor) [inline config/app.php:28]: nitpicky non-actionable thought: could be a separate commit on main too!

File: resources/views/endorsements/examiners.blade.php
Comment (thor) [inline resources/views/endorsements/examiners.blade.php:1]: Could these sections be a dedicated layout for us to import instead of duplicating? 👀 
image

**Review by thor — APPROVED** [review]
Comments ahoy! Nothing really blocking, but some docs suggestions and alternatives.

File: CONTRIBUTE.md [hunk 8]
```diff
@@ -22,8 +22,15 @@ If you encounter permissions errors you might want to `chown -R www-data:www-dat
...
 ### Run unit tests
 
+#### First run
+
+On first run you might need to setup the testing sqlite database first.
+Run the command `php artisan migrate --database sqlite-testing` to setup the database.
+
+#### Test
+
 To run the PHP unit tests use `./vendor/bin/phpunit --color=always --testdox`\
```
[the remark this answers is not in the document]
Comment (blt950) [reply CONTRIBUTE.md:32]: I wonder it was broken somehow, but I've added it instead. We'll quickly figure it out

File: CONTRIBUTE.md [hunk 9]
```diff
@@ -22,8 +22,15 @@ If you encounter permissions errors you might want to `chown -R www-data:www-dat
 
 ### Run unit tests
 
+#### First run
+
+On first run you might need to setup the testing sqlite database first.
+Run the command `php artisan migrate --database sqlite-testing` to setup the database.
```
[the remark this answers is not in the document]
Comment (blt950) [reply CONTRIBUTE.md:28]: No it won't, I tested this yesterday and you just get sql errors or something iirc. So we need to migrate the database, and I think it's best to use the laravel command for that instead of manual touching

File: README.md (hunk 2 above)
Comment (blt950) [reply README.md:36]: This will work for the key, but not artisan migrate, because migrate will ask you "are you sure, bc its prod" and docker exec doesn't support this and ends the execute after it's called and gets first reply. This gets more tricky with `php artisan user:makeadmin` that will ask multiple questions etc. Migrate prolly has some force argument, but makeadmin doesn't (yet).

I think we should look into this in a separate PR, because I agree docker exec would be even easier to use for most users. Though I don't want to mess with this 5 min before pushing this

File: resources/views/endorsements/examiners.blade.php
Comment (blt950) [reply resources/views/endorsements/examiners.blade.php:1]: We could maaaaaybe use a blade component for this, but that's a separate task :D

File: CONTRIBUTE.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (thor) [reply CONTRIBUTE.md:28]: Hmm. Are you sure? I just tested the same locally and it worked. All I needed to do was to create the database file.

File: README.md (hunk 2 above)
Comment (thor) [reply README.md:36]: That's a good point, but `docker exec` supports it: just needs `-it` instead of `-t`, no?

File: CONTRIBUTE.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (thor) [reply CONTRIBUTE.md:28]: Updated comment with the alternate suggestion, although I haven't been able to reproduce issues with testing with only `touch database/testing.sqlite` yet.

[the remark this answers is not in the document]
Comment (blt950) [reply CONTRIBUTE.md:28]: When I ran test yesterday it was without any database file, so it just errored. Though if you mean that touching the file and then running tests, then it's possible it works and auto-migrates.

Though I'm more fan of using the artisan commands to accomplish the same thing, because suddenly Laravel does a change in their test handling and then it's best to use the "standard way" of testing, rather than manually touching files imo.

File: README.md (hunk 2 above)
Comment (blt950) [reply README.md]: Good point, added in commit

File: UPGRADE.md (hunk 3 above)
Comment (blt950) [reply UPGRADE.md:21]: Done
